File: backend/src/main/java/com/peczedavid/fogorvos/model/db/Role.java
package com.peczedavid.fogorvos.model.db;

import lombok.*;

import javax.persistence.*;

@Data
@RequiredArgsConstructor
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "roles")
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NonNull
    @Column(nullable = false, unique = true)
    private String name;

    public static final String ROLE_USER = "ROLE_USER";
    public static final String ROLE_RECEPTIONIST = "ROLE_RECEPTIONIST";
    public static final String ROLE_ADMIN = "ROLE_ADMIN";

}

File: frontend/src/app/model/role.ts
export interface Role {
  id: string;
  name: string;
}

export const ROLE_USER = 'ROLE_USER';
export const ROLE_RECEPTIONIST = 'ROLE_RECEPTIONIST';
export const ROLE_ADMIN = 'ROLE_ADMIN';

export function mapRoleToReadableName(role: string): string {
  switch (role) {
    case ROLE_USER:
      return 'Felhasználó';
    case ROLE_RECEPTIONIST:
      return 'Recepciós';
    case ROLE_ADMIN:
      return 'Admin';
  }
  return 'Hibás szerepkör';
}